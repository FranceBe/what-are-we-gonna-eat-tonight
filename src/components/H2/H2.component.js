import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { fontSizes, spaces, colors } from 'styles/variables'

export const H2Default = styled.h2`
  font-size: ${fontSizes.px_20};
  text-align: center;
  text-shadow: ${spaces.xtraSmall} ${spaces.xtraSmall} ${colors.primary_default};
`

export class H2 extends React.PureComponent {
  static propTypes = {
    children: PropTypes.any,
  }

  render() {
    return (
      <H2Default>{this.props.children}</H2Default>
    )
  }
}

export default H2
