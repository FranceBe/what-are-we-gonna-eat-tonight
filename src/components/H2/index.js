import H2Component, { H2, H2Default } from 'components/H2/H2.component'

export { H2, H2Default }

export default H2Component
