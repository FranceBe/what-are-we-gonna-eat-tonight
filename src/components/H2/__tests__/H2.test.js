import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import H2, { H2Default } from 'components/H2'

describe('H2', () => {
  describe('Styled component', () => {
    describe('H2Default', () => {
      it('should match snapshot', () => {
        const H2DefaultTree = renderer.create(<H2Default/>).toJSON()
        expect(H2DefaultTree).toMatchSnapshot()
      })
      it('should have font-size set to 1.25rem', () => {
        const H2DefaultTree = renderer.create(<H2Default/>).toJSON()
        expect(H2DefaultTree).toHaveStyleRule('font-size', '1.25rem')
      })
      it('should have text-align set to center', () => {
        const H2DefaultTree = renderer.create(<H2Default/>).toJSON()
        expect(H2DefaultTree).toHaveStyleRule('text-align', 'center')
      })
      it('should have text-shadow set to 1px 1px #EBA789', () => {
        const H2DefaultTree = renderer.create(<H2Default/>).toJSON()
        expect(H2DefaultTree).toHaveStyleRule('text-shadow', '1px 1px #EBA789')
      })
    })
  })
  describe('rendering', () => {
    describe('default props', () => {
      const children = 'I am an H2 title :)'
      it('should render H2Default component with its children', () => {
        const H2Wrapper = shallow(<H2>{children}</H2>)
        expect(H2Wrapper.find(H2Default)).toHaveLength(1)
        expect(H2Wrapper.find(H2Default).text()).toBe('I am an H2 title :)')
      })
    })
  })
})
