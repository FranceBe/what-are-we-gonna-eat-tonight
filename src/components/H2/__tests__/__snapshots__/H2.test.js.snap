// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 Styled component H2Default should match snapshot 1`] = `
.c0 {
  font-size: 1.25rem;
  text-align: center;
  text-shadow: 1px 1px #EBA789;
}

<h2
  className="c0"
/>
`;
